src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -269,29 +269,34 @@ export class HttpConnection implements IConnection {
             return;
         }
 
+        const transportExceptions: any[] = [];
         const transports = negotiateResponse.availableTransports || [];
         for (const endpoint of transports) {
-            this.connectionState = ConnectionState.Connecting;
-            const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
-            if (typeof transport === "number") {
-                this.transport = this.constructTransport(transport);
-                if (!negotiateResponse.connectionId) {
-                    negotiateResponse = await this.getNegotiationResponse(url);
-                    connectUrl = this.createConnectUrl(url, negotiateResponse.connectionId);
-                }
-                try {
+            try {
+                this.connectionState = ConnectionState.Connecting;
+                const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
+                if (typeof transport === "number") {
+                    this.transport = this.constructTransport(transport);
+                    if (!negotiateResponse.connectionId) {
+                        negotiateResponse = await this.getNegotiationResponse(url);
+                        connectUrl = this.createConnectUrl(url, negotiateResponse.connectionId);
+                    }
                     await this.transport!.connect(connectUrl, requestedTransferFormat);
                     this.changeState(ConnectionState.Connecting, ConnectionState.Connected);
                     return;
-                } catch (ex) {
-                    this.logger.log(LogLevel.Error, `Failed to start the transport '${HttpTransportType[transport]}': ${ex}`);
-                    this.connectionState = ConnectionState.Disconnected;
-                    negotiateResponse.connectionId = undefined;
                 }
+            } catch (ex) {
+                this.logger.log(LogLevel.Error, `Failed to start the transport '${endpoint.transport}': ${ex}`);
+                this.connectionState = ConnectionState.Disconnected;
+                negotiateResponse.connectionId = undefined;
+                transportExceptions.push(`${endpoint.transport} failed: ${ex}`);
             }
         }
 
-        throw new Error("Unable to initialize any of the available transports.");
+        if (transportExceptions.length > 0) {
+            throw new Error(`Unable to connect to the server with any of the available transports. ${transportExceptions.join(" ")}`);
+        }
+        throw new Error("No transports are supported by the server.");
```

Comment:
To @halter73 's point, maybe

```suggestion
        throw new Error("None of the transports supported by the client are supported by the server.");
```

If you want to do that, also update the .NET client message :)